Saas broken unit tests schedule messed up

drigato opened the pull request:
Unit tests all pass now. They were failing because propel was returning a cached object collection.

Optimized the filling of linked show instances schedules:
When new show instances are created, we store their id in an associated array, organized by their cc_show id. This array gets passed to the fillNewLinkedInstances function and thus the guess work to determine which instances need their schedules filled is gone.

drigato committed:
Added criteria parameter to CcShow->getFutureInstanceIds so we can retrieve results from database instead of cache

drigato committed:
Better way of getting the linked show schedule and moved that into its own function.
Store the ids of the new, created show instances so we can pass them to the
fillNewLinkedInstances function

drigato committed:
Removed old comments

asantoni commented on airtime_mvc/application/models/airtime/CcShow.php:
```diff
@@ -281,9 +281,16 @@ public function getInstanceIds() {
+     * the current time
+     * 
+     * If a Criteria object is passed in Propel will always fetch the 
+     * results from the database and not return a cached collection
```
Is that true even if the criteria is NULL?

drigato replied:
No. It must be a criteria object; new Criteria(). I will add a comment

asantoni replied:
Ok... What's the normal way we want to use the function though? Should the default parameter by $criteria=Criteria()? 

(What's going to minimize programming errors in the future?)

asantoni commented on airtime_mvc/application/services/SchedulerService.php:
```diff
@@ -150,21 +150,53 @@ private static function findTimeDifference($p_startDT, $p_seconds)
+     */
+    public static function fillNewLinkedInstances($ccShow, $instanceIdsToFill)
+    {
+        //TODO can we remove the code until line 216 ??
```
???

Commit:
Clean up and simplifcation of linked show filling

asantoni commented on airtime_mvc/application/services/SchedulerService.php:
```diff
@@ -193,32 +225,27 @@ public static function fillNewLinkedInstances($ccShow)
-         * instance, which will have the most up to date schedule.
-         */
-       $showsPopulatedUntil = Application_Model_Preference::GetShowsPopulatedUntil();
+        $linkedShowSchedule = self::getLinkedShowSchedule($ccShow->getDbId(), $instanceIdsToFill);
```
This is way easier to understand now, good stuff.

asantoni commented on airtime_mvc/application/services/ShowService.php:
```diff
@@ -15,13 +15,18 @@ class Application_Service_ShowService
     private $repeatType;
     private $isUpdate;
-    private $linkedShowContent;
+    /*private $linkedShowContent;*/
```
Nuke it

asantoni commented on airtime_mvc/application/services/ShowService.php:
```diff
@@ -1229,11 +1247,11 @@ private function createMonthlyRepeatInstances($showDay, $populateUntil)
                     $newInstance = true;
                     $ccShowInstance = new CcShowInstances();
-                    $updateScheduleStatus = false;
+                    //$updateScheduleStatus = false;
```
Clean up these guys too please!

drigato committed:
Removed unused function.
Removed commented out code.
Renamed fillNewLinkedInstances to fillLinkedInstances.
Streamline show instance content if a show goes from unlinked to linked.

drigato committed:
Removed unused code.
Changed linked show warning message in the edit show form.

drigato committed:
Removed database test that confirms the is_linkable is set to false after a show is un-linked.
Since we removed this behaviour from Airtime, and a show can be linked and unlinked as many
times as the user wants, this test is deprecated.

drigato committed:
Added comments and removed some unused code
